core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -768,7 +768,12 @@ private void createOrderHash(QueryModel model) {
         if (n > 0) {
             final IntList orderByDirection = model.getOrderByDirection();
             for (int i = 0; i < n; i++) {
```

Comment:
It doesn't apply to qdb (yet) because order by accepts literals only but if sorting by expressions/functions becomes available this check might produce bad results for queries like : 

select * from t order by rnd_int() asc, rnd_int() desc;

Reply:
that's a good realization! 

I suspect once qdb supports expressions in the ORDER BY clause then the whole concept with a map `<column_name, direction>` will have to be re-evaluated. 

(fwiw: in this very example, it [wouldn't be easy](https://xkcd.com/221/) to prove it's broken :) but yes, there could be another example where it's more noticeable)